### PR TITLE
Populate genesis meta reveals

### PR DESCRIPTION
**Pull request by jharveyb**
Replacement PR for #458 to switch the author.

Tracking issue: https://github.com/lightninglabs/taproot-assets/issues/441

This PR includes some RPC changes not specified in that issue, but which were needed to get checks working in itest. The protos for proofs will change again in a later PR to include the group witness, and actual verification of these reveal fields will be implemented in a future PR.

### CODE REVIEW

**Review by halseth — APPROVED** [review]
Great, this is getting there!

File: itest/assets_test.go [hunk 1]
```diff
@@ -199,6 +199,7 @@ func mintAssetUnconfirmed(t *harnessTest, tapd *tapdHarness,
 	unconfirmedAssets := groupAssetsByName(listRespUnconfirmed.Assets)
 	for _, assetRequest := range assetRequests {
 		metaHash := (&proof.MetaReveal{
+			Type: proof.MetaOpaque,
```
Comment (Roasbeef) [inline itest/assets_test.go:202]: IIUC, this was the existing default value, so what does this change add?

File: taprpc/taprootassets.proto [hunk 2]
```diff
@@ -236,6 +236,24 @@ message AssetGroup {
     specified by the above key.
     */
     bytes asset_witness = 3;
+
+    // TODO(jhb): update to include tapscript_root
```
Comment (Roasbeef) [inline taprpc/taprootassets.proto:240]: Why's this need the root, wouldn't that instead be in `GroupKeyReveal` only? The witness would include the information needed to arrive at the root along with the tapscript leaf being revealed.

File: itest/assets_test.go (hunk 1 above)
Comment (guggero) [reply itest/assets_test.go:202]: No, the `proof.MetaOpaque` is value `1`. But on the RPC it's `0`. So we didn't have it translated properly between native and RPC types.

**Review by guggero — APPROVED** [review]
LGTM :tada:

File: taprpc/taprootassets.proto (hunk 2 above)
Comment (jharveyb) [reply taprpc/taprootassets.proto:240]: True, will remove in follow-up.

**Pull request merged**